mailman/mailman.go

```
@@ -70,19 +75,28 @@ func (m *Mailman) tokenState(c *aries.C) (*oauth2.Token, *oauth.State, error) {
 }
 
 func (m *Mailman) serveIndex(c *aries.C) error {
-	tok, err := m.token.Get(c.Context)
+	emails, ok := c.Req.URL.Query()["email"]
+	if !ok {
+		c.Resp.Write([]byte("Please specify email parameter"))
```

please return an error here:

`return errcode.InvalidArgf("email not specified in query")`

---

the error message will be printed to Resp's body.

also, next time you want to print a string to a Writer, use `io.WriteString()`

---

ok